test/tests/performance/pbs_preemptperformance.py

```diff
@@ -399,6 +399,81 @@ def test_insufficient_resc_multiple_non_cons(self):
         self.logger.info('#' * 80)
         self.logger.info('#' * 80)
 
+    @timeout(3600)
+    @tags('sched', 'scheduling_policy')
+    def test_insufficient_server_resc(self):
+        """
+        Submit a number of low priority jobs and then make the last low
+        priority job to consume some server level resources. Submit a
+        high priority job that request for this server level resource
+        and measure the time it takes for preemption.
+        """
+
+        a = {'type': 'long', 'flag': 'q'}
+        self.server.manager(MGR_CMD_CREATE, RSC, a, id='foo')
+
+        a = {ATTR_rescavail + ".ncpus": "8"}
+        self.server.create_vnodes(
+            "vn1", a, 401, self.mom, additive=True, fname="vnodedef1")
+
+        # Make resource foo available on server
+        a = {ATTR_rescavail + ".foo": 50, 'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        a = {ATTR_l + '.select': '1:ncpus=1'}
+        for _ in range(3200):
+            j = Job(TEST_USER, attrs=a)
+            j.set_sleep_time(3000)
+            self.server.submit(j)
+
+        # Add foo to the resources scheduler checks for
+        self.scheduler.add_resource('foo')
+        self.scheduler.unset_sched_config('preempt_sort')
+
+        a = {ATTR_l + '.select': '1:ncpus=1', ATTR_l + '.foo': 25}
+        j = Job(TEST_USER, attrs=a)
+        j.set_sleep_time(3000)
+        jid = self.server.submit(j)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {'substate=42': 3200}, interval=20,
+                           offset=15)
+
+        qname = 'highp'
+        a = {'queue_type': 'execution', 'priority': '200',
+             'started': 'True', 'enabled': 'True'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, qname)
+
+        a = {ATTR_l + '.select': '1:ncpus=1', ATTR_l + '.foo': 50,
+             ATTR_q: 'highp'}
+        j2 = Job(TEST_USER, attrs=a)
+        j2.set_sleep_time(3000)
+        jid_highp = self.server.submit(j2)
+
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid_highp, interval=10)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid)
+        search_str = jid_highp + ";Considering job to run"
+        (_, str1) = self.scheduler.log_match(search_str,
```

Comment:
You are right, but I am just paranoid about timing. If the match is possible then it will find it in the first attempt as well. If you want I can remove max_attempts itself.

Reply:
The job literally cannot go in the 'R' state before the scheduler logs those messages, so I think it should be fairly safe to make max_attempts = 1.

Reply:
I know, I'm just being irrational here. I'll change it.

Reply:
Well, there isn't a race here as such, since the job is running a log message must also exist. Ravi is right, it isn't needed here.